subprojects/docs/src/docs/release/notes.md

```diff
@@ -274,7 +274,7 @@ This flag may be useful in CI environments to record Gradle version information
 
 #### Checkstyle tasks use toolchains and execute in parallel by default
 
-The [Checkstyle plugin](userguide/checkstyle_plugin.html) now uses the Gradle worker API to run Checkstyle as an external worker process, so that multiple Checkstyle tasks may now run in parallel within a project. This can greatly increase overall build performance when several of these tasks exist within a single project. The memory used by the process is controlled via the `minHeapSize` and `maxHeapSize` properties.
+The [Checkstyle plugin](userguide/checkstyle_plugin.html) now uses the Gradle worker API to run Checkstyle as an external worker process, so that multiple Checkstyle tasks may now run in parallel within a project. This can greatly increase overall build performance when several of these tasks exist within a single project. In case of memory issues, you can adjust the memory used by the Checkstyle process via the `minHeapSize` and `maxHeapSize` properties on the Checkstyle task. By default the process will start with `maxHeapSize` of 512MB. We also recommend to use Checkstyle 9.3 or later.
```

Comment:
For info about fixing potential issues with this change, I would instead link to the upgrade guide.

Reply:
Removed